mesonbuild/interpreter.py

```diff
@@ -3742,6 +3754,9 @@ def func_configure_file(self, node, args, kwargs):
             # that the command is run from is 'unspecified', so it could change.
             # Currently it's builddir/subdir for in_builddir else srcdir/subdir.
             values = mesonlib.get_filenames_templates_dict(inputs_abs, [ofile_abs])
+            if depfile:
+                depfile = os.path.join(self.environment.build_dir, depfile)
```

Comment:
This should be `self.environment.get_scratch_dir()` rather than `self.environment.build_dir`.

Reply:
make sense, done